src/main/java/com/hivemq/cli/mqtt/ContextClientDisconnectListener.java

```diff
@@ -19,17 +19,23 @@
 import com.hivemq.cli.commands.shell.ShellCommand;
 import com.hivemq.cli.commands.shell.ShellContextCommand;
 import com.hivemq.cli.utils.MqttUtils;
+import com.hivemq.client.mqtt.MqttClientConfig;
 import com.hivemq.client.mqtt.lifecycle.MqttClientDisconnectedContext;
 import com.hivemq.client.mqtt.lifecycle.MqttClientDisconnectedListener;
 import com.hivemq.client.mqtt.lifecycle.MqttDisconnectSource;
 import org.jetbrains.annotations.NotNull;
 import org.pmw.tinylog.Logger;
+import org.pmw.tinylog.LoggingContext;
 
 public class ContextClientDisconnectListener implements MqttClientDisconnectedListener {
 
     @Override
     public void onDisconnected(final @NotNull MqttClientDisconnectedContext context) {
 
+        final String contextBefore = LoggingContext.get("identifier");
+
+        LoggingContext.put("identifier", "CLIENT " + context.getClientConfig().getClientIdentifier().orElse(null));
+
         if (context.getSource() == MqttDisconnectSource.SERVER) {
```

Comment:
Why is the context only removed if the server closed the connection? Does not make sense to me.

Reply:
Good catch! Thanks 👍 
My thoughts were to only handle closed connections from the server side as an error because I didn't know that e. g. timeouts are client side failures.
I fixed this in recent commits so that every closed connection gets handled as error except for user caused errors.